src/mon/OSDMonitor.cc

```diff
@@ -6832,6 +6871,9 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 	return true;
       }
       int id = newcrush.get_item_id(name);
+      int ancestor = 0;
+      bool has_ancestor = false;
```

Comment:
fwiw this bool is redudnant; any ancestor will be a bucket with a value <0, so you can take 0 to mean 'no ancestor'.